internal/gengapic/genrest.go

```diff
@@ -714,23 +722,35 @@ func (g *generator) unaryRESTCall(servName string, m *descriptor.MethodDescripto
 	// TODO(dovs): handle http headers
 	// TODO(dovs): handle deadlines?
 	// TODO(dovs): handle calloptions
-	p("m := protojson.MarshalOptions{AllowPartial: true, EmitUnpopulated: true}")
-	requestObject := "req"
-	if info.body != "" && info.body != "*" {
-		requestObject = "body"
-		p("body := req.Get%s()", snakeToCamel(info.body))
+
+	body := "nil"
+
+	// Marshal body for HTTP methods that take a body.
```

Comment:
Suggest adding a "// TODO: Have a test check that we generate requests with and without a body", since it looks like no test files are changing because of this fix.

Reply:
Yep. Done.